hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/TransferPrecompile.java

```diff
@@ -562,6 +578,26 @@ private List<BalanceChange> constructBalanceChanges(final CryptoTransferWrapper
                 }
             }
 
+            for (var hbarTransfer : transferOp.transferWrapper().hbarTransfers()) {
+                if (hbarTransfer.sender() != null) {
+                    changes.add(
+                            BalanceChange.changingHbar(
+                                    aaWith(
+                                            hbarTransfer.sender(),
+                                            -hbarTransfer.amount(),
+                                            hbarTransfer.isApproval()),
+                                    EntityIdUtils.accountIdFromEvmAddress(senderAddress)));
+                } else if (hbarTransfer.receiver() != null) {
+                    changes.add(
+                            BalanceChange.changingHbar(
+                                    aaWith(
+                                            hbarTransfer.receiver(),
+                                            hbarTransfer.amount(),
+                                            hbarTransfer.isApproval()),
+                                    EntityIdUtils.accountIdFromEvmAddress(senderAddress)));
+                }
+            }
+
```

Comment:
⚠️  The hbar transfers loop is currently nested in the token transfers loop, it should be extracted, otherwise hbar balance changes may be repeated multiple times or not even acknowledged if there are no token transfers.

Reply:
yikes! thanks for catching this.  Had it right in my other repo but clearly this is wrong the way it is.